### PR TITLE
TBB 2019.0 with VS2008 fix

### PR DESCRIPTION
**Pull request by anton-malakhov**
Updating to TBB version 2019.0 along with the following changes in the recipe:
* versioning scheme changed along with the rest of the Intel Parallel Studio packages
* python-tbb entry point removed (it behaves differently from `python -m tbb`)
* added temporary work-around patch for compilation problem on VS2008 (not officially supported by TBB team anymore)  
* added compilation test for `tbb-devel` instead of simple checks for files existence
* added `run_export` section and `make` as a build dependency on [unix]

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

Replaces #34

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -52,28 +55,38 @@ outputs:
...
+        - examples/test_all/fibonacci
       commands:
-        - if not exist %PREFIX%\Library\include\tbb\tbb.h exit 1  # [win]
-        - if not exist %PREFIX%\Library\lib\tbb.lib exit 1        # [win]
-        - test -f ${PREFIX}/include/tbb/tbb.h                     # [unix]
+        - cd examples\test_all\fibonacci                 # [win]
+        - set INCLUDE=%INCLUDE%;%LIBRARY_INC%            # [win]
+        - set LIB=%LIB%;%LIBRARY_LIB%                    # [win]
+        - nmake /f Makefile.windows                      # [win]
+        - make -C examples/test_all/fibonacci CXXFLAGS="-I$PREFIX/include -L$PREFIX/lib"  # [unix]
```
Comment (anton-malakhov) [inline recipe/meta.yaml]: The fact that conda-build requires include and library directories settings to be explicit was unexpected. Am I doing something wrong with the compiler in the test requirements?

Comment (isuruf) [reply recipe/meta.yaml]: The Makefile doesn't make use of CPPFLAGS and uses only CXXFLAGS. Toolchain adds `-I$PREFIX/include` to CPPFLAGS and not CXXFLAGS.

Comment (anton-malakhov) [reply recipe/meta.yaml]: Thanks, I see. What's about the same question on Windows?

Comment (isuruf) [reply recipe/meta.yaml]: INCLUDE and LIB are not set in https://github.com/AnacondaRecipes/aggregate/blob/master/vs2015/activate.bat . You'll have to open an issue in that repo asking them to set INCLUDE and LIB

Comment (anton-malakhov) [reply recipe/meta.yaml]: Thanks! Also, w.r.t. `CPPFLAGS` which is the pre-processor flags only.. what's supplying `-L` flag or sets `LIBRARY_PATH`?

Comment (isuruf) [reply recipe/meta.yaml]: `LDFLAGS` should have `-L$PREFIX/lib`

**Comment by anton-malakhov** [comment]
The build is green now! However, I've noticed a problem with debug libraries within `tbb-devel` package #36 . I'll look into this tomorrow

**Comment by anton-malakhov** [comment]
It is ready for merge: builds are green and the issue with the debug binaries is worked-around.

**Pull request merged**